pkg/derivation/derivation.go

```diff
@@ -81,29 +86,163 @@ func (d *Derivation) Validate() error {
 	return nil
 }
 
+func compressHash(hash []byte, newSize int) []byte {
+	buf := make([]byte, newSize)
+	for i := 0; i < len(hash); i++ {
+		buf[i%newSize] ^= hash[i]
+	}
+	return buf
+}
+
+func (d *Derivation) Name() string {
+	for _, e := range d.EnvVars {
+		if e.Key == "name" {
+			return e.Value
+		}
+	}
+
+	// TODO: Maybe panic or change type sig to (string, error)?
+	return ""
```

Comment:
`panic` here, and return an error in all parser codepaths - which in our case, would be best in the `Validate()` function (so people populating structs manually could still check if through `Validate()` without being afraid of panics)

Reply:
Ah, and add a comment next to the panic statement that this is verified in the `Validate` method, so can't really happen on validated inputs.